__tests__/app.test.js

```diff
@@ -105,6 +105,44 @@ describe("/api/articles", () => {
         });
       });
   });
+  test("GET:200 Topic query filters the articles based on topic", () => {
+    return request(app)
+      .get("/api/articles?topic=mitch")
+      .expect(200)
+      .then(({ body }) => {
+        expect(body.articles.length).toBe(12);
+        body.articles.forEach((article) => {
+          expect(typeof article.author).toBe("string");
+          expect(typeof article.title).toBe("string");
+          expect(typeof article.article_id).toBe("number");
+          expect(typeof article.topic).toBe("string");
+          expect(typeof article.created_at).toBe("string");
+          expect(typeof article.votes).toBe("number");
+          expect(typeof article.article_img_url).toBe("string");
+          expect(typeof article.total_comments).toBe("string");
+        });
+        expect(body.articles).toBeSortedBy("created_at", {
+          coerce: true,
+          descending: true,
+        });
+      });
+  });
+  test("GET:200 Topic query filters to an empty array when given a valid but non-existant query", () => {
+    return request(app)
+      .get("/api/articles?topic=nonexistanttopic")
```

Comment:
This - `nonexistanttopic` -  is not a valid topic as it is not in the data base. Have a look in the test data for a topic with no articles and use that for this test (the assertion is correct  though - we want that empty array back)